lib/sidebar.rb

```diff
@@ -3,211 +3,131 @@
-                    else
-                      "<li class=\"nav-leaf\"><a href=\"#{value[:url]}\">#{title}</a></li>"
-                    end
       end
 
-      subgroup
-    end
-
-    def render(filename)
-      sidebar = ''
+      def find_headers(doc)
+        headers = []
 
-      merged = merge(@hash_pre_render, @filename_with_headers).sort_by { |_key, value| value[:menu_order] }
-      merged.select { |key, _value| key.split('/').length <= 2 }.each do |key, value|
-        next if value[:title].nil?
-        next if value[:hide_from_sidebar]
+        doc.xpath('//h2').each do |header|
+          next unless header['id']
 
-        subgroups = merged.select { |subgroup_key, _subgroup_value| subgroup_key.include? key and subgroup_key != key and key != '/' }
+          child_markup = header.last_element_child
+          header = {
+            id: header['id'],
+            title: header.content.strip,
+            hash: (child_markup['href']).to_s
+          }
+          headers.push(header)
+        end
 
-        active = active?(filename, key)
-        # puts "#{filename}, #{key}" if active
-        item_class = active ? 'nav-group active' : 'nav-group'
+        headers
+      end
 
-        child = "<li class=\"#{item_class}\">"
-        child << "<div class=\"nav-group-heading\"><i class=\"material-icons\">arrow_right</i><span>#{value[:title].split('–').first}</span></div>"
+      def write_sidebar_to_files(files)
+        files.each do |file|
+          sanitized_filename = file[:sanitized_filename]
+          filename = file[:filename]
+          doc = file[:doc]
+          sidebar_markup = render(sanitized_filename, pages)
 
-        child << '<ul class="nav-ul">'
+          next if sidebar_markup.nil?
 
-        subgroup = generateSubgroup(filename, key, value, subgroups, 2)
+          sidebar_containers = find_containers(doc, filename)
 
-        child << subgroup
+          next if sidebar_containers.nil?
 
-        if subgroups.any?
-          subgroups.select { |subgroup_key, _subgroup_value| subgroup_key.split('/').length <= 3 }.each do |subgroup_key, subgroup_value|
-            subgroup = generateSubgroup(filename, subgroup_key, subgroup_value, subgroups, 2)
-            child << subgroup
+          sidebar_containers.each do |sidebar_container|
+            sidebar_container.inner_html = sidebar_markup
           end
-        end
-
-        child << '</ul>'
-        child << '</li>'
-        sidebar << child
-      end
-
-      File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
-      sidebar
-    end
 
-    def active?(filename, url, exact = false)
-      if filename == '/' || url == '/'
-        if filename == '/' && url == '/'
-          return true
-        else
-          return false
+          Jekyll.logger.debug("   Writing Sidebar: #{filename}")
+          File.open(filename, 'w') { |f| f.write(doc.to_html(encoding: 'UTF-8')) }
```

Comment:
Codacy found an issue: [Line is too long. [82/80]](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)